### PR TITLE
Dispense Medicine Feature

### LINKED ISSUE #129: Assign Medicine Feature
Assigns medication to Patients on the day of appointment. Returns batch id of medication to dispense to Patient.

### PR DESCRIPTION
**Pull request by shxr3f**
Implemented feature to dispense medicine to patient who has appointment today.
Included new storage componnent that saves all the medicines given in an appointment.
This resolves #129 
Please do take a look at the code implemented. Do ask if you have any doubts on how it was implemented!

### CODE REVIEW

File: src/main/java/seedu/duke/Manager.java [hunk 1]
```diff
@@ -138,6 +138,17 @@ private Status executeCommand(String commandWord, String parameters) throws Halp
...
+        case "dispense medicine":
+            command = Parser.parseDispenseMedicine(parameters);
+
+            Command helperCommand = Parser.parseCheckForPatientAppointment(parameters);
+            helperCommand.execute(storage.appointments);
+            helperCommand = Parser.parseCheckMedicineStock(parameters);
+            helperCommand.execute(storage.medicines);
+
+            command.execute(storage.appointments);
+            status = command.execute(storage.medicines);
```
Comment (wraineflores) [inline src/main/java/seedu/duke/Manager.java:150]: Maybe extract this as a method if it's possible? so that our case switch isn't as long

**Review by wraineflores — COMMENTED** [review]
Extract case switch

File: src/main/java/seedu/duke/Manager.java (hunk 1 above)
Comment (tanweili) [reply src/main/java/seedu/duke/Manager.java:150]: I think this can be done in another PR and account for the other commands as well.

**Review by tanweili — APPROVED** [review]
I think the coding style wise does fulfill coding standard and our code structure. I think can merge

**Pull request merged**